### PR TITLE
Store rotational histogram as part of the node data.

### PR DESCRIPTION
**Pull request by wohe**
The compressed point cloud is no longer stored as part of the
node data to reduce memory consumption.

### CODE REVIEW

File: cartographer/mapping/map_builder.cc [hunk 1]
```diff
@@ -181,9 +181,7 @@ void MapBuilder::SerializeState(io::ProtoStreamWriter* const writer) {
         // TODO(whess): Handle trimmed data.
         range_data_proto->mutable_node_id()->set_trajectory_id(trajectory_id);
         range_data_proto->mutable_node_id()->set_node_index(node_index);
-        const auto& data = *node_data[trajectory_id][node_index].constant_data;
-        *range_data_proto->mutable_range_data() =
-            sensor::ToProto(data.range_data);
+        range_data_proto->mutable_range_data();
```
Comment (cschuet) [inline cartographer/mapping/map_builder.cc]: I don't understand. Is this not a no-op?

File: cartographer/mapping/trajectory_node.h [hunk 2]
```diff
@@ -46,6 +43,7 @@ struct TrajectoryNode {
     // Used for loop closure in 3D.
     sensor::PointCloud high_resolution_point_cloud;
     sensor::PointCloud low_resolution_point_cloud;
+    Eigen::VectorXf rotational_scan_matcher_data;
```
Comment (cschuet) [inline cartographer/mapping/trajectory_node.h]: Mention that this is histogram data?

File: cartographer/mapping_3d/local_trajectory_builder.h [hunk 3]
```diff
@@ -68,6 +68,7 @@ class LocalTrajectoryBuilder {
       const Eigen::Quaterniond& gravity_alignment,
       const sensor::PointCloud& high_resolution_point_cloud,
       const sensor::PointCloud& low_resolution_point_cloud,
+      const Eigen::VectorXf& rotational_scan_matcher_data,
```
Comment (cschuet) [inline cartographer/mapping_3d/local_trajectory_builder.h]: histogram?

File: cartographer/mapping/map_builder.cc (hunk 1 above)
Comment (wohe) [reply cartographer/mapping/map_builder.cc]: We don't have any range data anymore and I would like to send a follow up PR for serializing the new data to keep this PR small and focused. This is adding the proto message without adding any data in it. So strictly speaking not a no-op.

**Pull request merged**